tempesta_fw/t/functional/tests/test_frang.py

```diff
@@ -0,0 +1,516 @@
+		if len(data) == 0:
+			self.res = True
+			tcount += 1
+
+		print("res:", self.res)
+		self.s.close()
+		time.sleep(5)
+		tfw.stop()
+
+	def body_len(self):
+		global tcount
+
+		print("body_len")
+		self.res = False
+		self.__init__()
+		self.cfg.add_section('frang_limits')
+		self.cfg.add_option('http_ct_vals', '[\"text/html\"]')
+		self.cfg.add_option('http_body_len', '10')
+		self.cfg.add_end_of_section()
+		self.vs_get = b"POST /a.html HTTP/1.0\r\nHost: loc\r\n" +\
+b"Content-Type: text/html\r\n" +\
+b"Content-Length: 20\r\n\r\n" +\
+b"<html>content</html>\r\n\r\n"
+		tfw.start_with_frang()
+		print("tfw start\n")
+		self.s = socket(AF_INET, SOCK_STREAM)
+		self.s.connect(('127.0.0.1', 8081))
+		self.s.send(self.vs_get)
+		data = self.s.recv(1024)
+		if len(data) == 0:
+			self.res = True
+			tcount += 1
+
+		print("res:", self.res)
+		self.s.close()
+		time.sleep(5)
+		tfw.stop()
+
+	def field_len(self):
+		global tcount
+
+		print("field_len")
+		self.res = False
+		self.__init__()
+		self.cfg.add_section('frang_limits')
+		self.cfg.add_option('http_ct_vals', '[\"text/html\"]')
+		self.cfg.add_option('http_field_len', '10')
+		self.cfg.add_end_of_section()
+		self.vs_get = b"POST /a.html HTTP/1.0\r\nHost: loc\r\n" +\
+b"Content-Type: application/xml\r\n" +\
+b"Content-Length: 20\r\n\r\n" +\
+b"<html>content</html>\r\n\r\n"
+		tfw.start_with_frang()
+		print("tfw start\n")
+		self.s = socket(AF_INET, SOCK_STREAM)
+		self.s.connect(('127.0.0.1', 8081))
+		self.s.send(self.vs_get)
+		data = self.s.recv(1024)
+		if len(data) == 0:
+			self.res = True
+			tcount += 1
+
+		print("res:", self.res)
+		self.s.close()
+		time.sleep(5)
+		tfw.stop()
+
+	def ct_vals(self):
+		global tcount
+
+		print("ct_vals")
+		self.res = False
+		self.__init__()
+		self.cfg.add_section('frang_limits')
+		self.cfg.add_option('http_ct_vals', '[\"text/html\"]')
+		self.cfg.add_end_of_section()
+		self.vs_get = b"POST / HTTP/1.0\r\nhost: loc\r\n" +\
+b"Content-type: application/xml\r\n\r\n"
+		tfw.start_with_frang()
+		print("tfw start\n")
+		self.s = socket(AF_INET, SOCK_STREAM)
+		self.s.connect(('127.0.0.1', 8081))
+		self.s.send(self.vs_get)
+		data = self.s.recv(1024)
+		if len(data) == 0:
+			self.res = True
+			tcount += 1
+
+		print("res:", self.res)
+		self.s.close()
+		time.sleep(5)
+		tfw.stop()
+
+	def req_method(self):
+		global tcount
+
+		print("req_method")
+		self.res = False
+		self.__init__()
+		self.cfg.add_section('frang_limits')
+		self.cfg.add_option('http_methods', 'get')
+		self.cfg.add_end_of_section()
+		self.vs_get = b"POST / HTTP/1.0\r\nhost: loc\r\n" +\
+b"Content-type: application/xml\r\n\r\n"
+		tfw.start_with_frang()
+		print("tfw start\n")
+		self.s = socket(AF_INET, SOCK_STREAM)
+		self.s.connect(('127.0.0.1', 8081))
+		self.s.send(self.vs_get)
+		data = self.s.recv(1024)
+		if len(data) == 0:
+			self.res = True
+			tcount += 1
+
+		print("res:", self.res)
+		self.s.close()
+		time.sleep(5)
+		tfw.stop()
```

Comment:
No empty line before the method.

Also please write good comments for each method what the method verifies and how the verification works.